dongle_app/libs/fox_player.js

```diff
@@ -20,19 +20,14 @@ var FoxPlayer = (function(){
   }
 
   function close(){
+    document.mozCancelFullScreen();
```

Comment:
Here you can have just `window.close()` and that's all. This will remove all the stuff related (full screen among others). So no need to wait for closing.